arroyo/processing/strategies/produce.py

```diff
@@ -0,0 +1,86 @@
+import logging
+import time
+from collections import deque
+from concurrent.futures import Future
+from typing import Deque, Optional, Tuple
+
+from arroyo.backends.abstract import Producer
+from arroyo.processing.strategies.abstract import MessageRejected, ProcessingStrategy
+from arroyo.types import Commit, Message, Position, Topic, TPayload
+
+logger = logging.getLogger(__name__)
+
+
+class ProduceAndCommit(ProcessingStrategy[TPayload]):
+    """
+    For each message received, produces a message to the topic being passed in.
+    Waits on the futures and only once completed, commits the offset of the message.
+    """
+
+    def __init__(
+        self,
+        producer: Producer[TPayload],
+        topic: Topic,
+        commit: Commit,
+        max_buffer_size: int = 10000,
+    ):
+        self.__producer = producer
+        self.__topic = topic
+        self.__commit = commit
+        self.__max_buffer_size = max_buffer_size
+
+        self.__queue: Deque[
+            Tuple[Message[TPayload], Future[Message[TPayload]]]
+        ] = deque()
+
+        self.__closed = False
+
+    def poll(self) -> None:
+        while self.__queue:
+            message, future = self.__queue[0]
+
+            if not future.done():
+                break
+
+            self.__queue.popleft()
+
+            self.__commit(
+                {message.partition: Position(message.next_offset, message.timestamp)}
+            )
```

Comment:
This will require a commit policy to be set on the stream processor in order not to commit every offset on high throughput consumers, which are most. 
Do you have a plan to ensure people who use this producer do set a proper commit policy ?

Reply:
No, there is no plan for this nor should there be. The commit policy design aims to make commit policy and strategies completely orthogonal concerns. All policies work with all strategies. Strategies should call commit as frequently as they can (i.e. on every message like this one does) and it's the sole job of the commit policy to decide when to stage vs commit.

If you are worried about people picking the wrong default, I would be open to changing the default policy to once per second instead of on every message (The only reason I did not do that previously was to make commit policies a no-op when first introduced)

Reply:
We set a default to make the whole thing a no-op and not require rewriting consumers when this was first introduced. I am happy to make it required in a future release and not have a default.

Reply:
Ok: Made the change here https://github.com/getsentry/arroyo/pull/116. Will update all the usages within our codebases before merging this.